doc/api/net.md

```diff
@@ -620,6 +620,18 @@ is received. For example, it is passed to the listeners of a
 [`'connection'`][] event emitted on a [`net.Server`][], so the user can use
 it to interact with the client.
 
+### Class property: `net.Socket.autoDetectFamily`
```

Comment:
I'm wary of non-`const` global properties that affect the entire application, e.g., most such properties turned out to be a mistake in `node:crypto`.

Reply:
+1 on this feedback. It's hard for me to think of a place where we've done this and it hasn't been an unwelcome foot-gun.

Reply:
I'm not fan of this as well actually.
I had to introduce otherwise most tests in the CI would fail as they use the `connection` event and `common.mustCall`. Which will fail with happy eyeballs attempts two connections.

Do you have a better idea?

Reply:
I actually like this. It's a one-line-fix for most IPv6 issues if we decide to not enable happy eyeballs by default.

Reply:
So shall I leave it or not?

Reply:
@mcollina Wouldn't an environment variable make more sense then if your intention is to quickly switch it on for an existing application? A global property requires changes to the program code and it cannot be set by libraries because that could break other components (which is why I am against global properties).

Reply:
(Environment variables are jut as bad as global variables, of course, and it makes it more difficult to change the behavior from within the application.)

Reply:
Yes, I can definitely go for an environment variable if it is an approach we accept.